packages/cli/lib/validators/marketplaceValidators/theme/ModuleValidator.js

```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const BaseValidator = require('../BaseValidator');
+const { isModuleFolderChild } = require('@hubspot/cli-lib/modules');
+
+const MODULE_LIMIT = 50;
+
+class ModuleValidator extends BaseValidator {
+  constructor(options) {
+    super(options);
+
+    this.errors = {
+      LIMIT_EXCEEDED: {
+        key: 'limitExceeded',
+        getCopy: ({ limit, total }) =>
+          `Cannot exceed ${limit} modules in your theme (found ${total})`,
+      },
+      MISSING_META_JSON: {
+        key: 'missingMetaJSON',
+        getCopy: ({ modulePath }) =>
+          `Missing a meta.json file for ${modulePath}`,
+      },
+      INVALID_META_JSON: {
+        key: 'invalidMetaJSON',
+        getCopy: ({ modulePath }) =>
+          `Invalid json in meta.json file for ${modulePath}`,
+      },
+      MISSING_LABEL: {
+        key: 'missingLabel',
+        getCopy: ({ modulePath }) =>
+          `The meta.json file is missing a "label" field for ${modulePath}`,
+      },
+      MISSING_ICON: {
+        key: 'missingIcon',
+        getCopy: ({ modulePath }) =>
+          `The meta.json file is missing an "icon" field for ${modulePath}`,
+      },
+    };
+  }
+
+  getUniqueModulesFromFiles(files) {
```

Comment:
This parses the file path list and picks out all of the unique modules in it. It returns an object that looks like this:
```js
const uniqueModules = {
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module.module': {
    'fields.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module.module/fields.json',
    'meta.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module.module/meta.json',
    'module.html': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module.module/module.html'
  },
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module': {
    'fields.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module/fields.json',
    'meta.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module/meta.json',
    'module.css': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module/module.css',
    'module.html': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module/module.html',
    'module.js': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module/module.js'
  },
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module': {
    'fields.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module/fields.json',
    'meta.json': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module/meta.json',
    'module.css': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module/module.css',
    'module.html': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module/module.html',
    'module.js': '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module/module.js'
  }
}
```

Reply:
Very cool!

Reply:
Would it be easier to store this as an array of the relative paths of the files? It looks like we are manipulating these to be that anyway later.
```
const uniqueModules = {
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module.module': [
    'fields.json',
    'meta.json',
    'module.html'
  ],
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_1.module': [
    'fields.json',
    'meta.json',
    'module.css',
    'module.html',
    'module.js'
  ],
  '/Users/brodgers/src/cms-devex-super-repo/hubspot-cli/tmp/my_module_2.module': [
    'fields.json',
    'meta.json',
    'module.css',
    'module.html',
    'module.js'
  ]
}
```